packages/autorest.go/src/generator/helpers.ts

```diff
@@ -575,6 +575,13 @@ export async function getParentImport(session: Session<CodeModel>): Promise<stri
   const modName = await session.getValue('module', 'none');
   const containingMod = await session.getValue('containing-module', 'none');
   if (modName !== 'none') {
+    const majorVersion = (await session.getValue('module-version', '1.0.0')).split('.')[0];
+    if (Number(majorVersion) > 1) {
+      const version = modName.substring(`github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/${clientPkg.substring(3)}/${clientPkg}`.length).match(/v\d/)?.[0];
```

Comment:
Is this just something for testing out a change?  We shouldn't have hard-coded module identities.

Reply:
@Alancere you should not hard code the modName, you just need to judge whether the last part of modName is vX.

Reply:
@jhendrixMSFT for control plane package, we set module-name in swagger repo's config. we do not want to change it for every major version release. @Alancere is helping to add a logic to add the suffix `vX` to the module name.

Reply:
So in the config, we'll have the following bits of data.
```
module: github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/foo/armfoo
module-version: v1.2.3
```
Are you saying that when `armfoo` goes to `v2.0.0` that only the `module-version` info is updated?

Reply:
Yes. Will it have any side effects for data plane?

Reply:
@jhendrixMSFT could you help to review again?